\title{Fortnightly report}
\author{
        Adam Kurkiewicz \\
       	College of Medical, Veterinary and Life Sciences\\
        University of Glasgow\\
	\href{mailto:a.kurkiewicz.1@research.gla.ac.uk}{a.kurkiewicz.1@research.gla.ac.uk}
}

\date{\today}

\documentclass[12pt]{article}

\usepackage{hyperref}
\usepackage{url}
\usepackage[
    backend=biber,
    style=authoryear-icomp,
    natbib=true,
    url=false, 
    doi=true,
    eprint=false
]{biblatex}
\usepackage{graphicx}
\usepackage[section]{placeins}
%\usepackage{float}

\addbibresource{main.bib}
\addbibresource[datatype=ris]{irizarry.ris}

\begin{document}
\maketitle

\section{Data}

I've finished cleaning up, assembling, filtering \& normalising the data. A single tab-separated file is now available in our google drive in a folder Data/KurkiewiczData~\parencite{GoogleDrive}. I will refer to this dataset as cleanMD.

As you may remember from last week, cleanMD contains 35 observations and over 20,000 variables. Variables include:
\begin{itemize}
  \item Modal Allele length (as measured by Darren's group).
  \item Progenial Allele length (estimated from Modal Allele Length).
  \item Disease status.
  \item RMA normalised $log_{2}$-transformed and aggregated into ``core'' level probe intensities. Each grouping of probes at the core level has a $transcript\_cluster\_id$, which can be uniquely mapped onto a known gene description/ gene id in a publically accessible database.
\end{itemize}

I am planning to include additional information in cleanMD, when required, by pulling more data points from various spreadsheets that were supplied and live now in our google drive.

\section{Normalisation}

I've tried to follow up on Simon's query regarding the precise method by which the aggregation/normalisation of probe intensities into transcript clusters happens. I've found a short paper by \cite{fRMA}, which explains what's going on. I'm still digesting it.

I've followed John's advise and I've produced boxplots of both transcript cluster intensities in cleanMD and ``gene exprssion levels'' from the dataset supplied by GeneLogic. The plots seem to show that oligo's RMA \ref{cleanMDBoxplot} did a far better job than GeneLogic's normalisation, although it isn't perhaps a fair comparison, because I didn't exclude muscle data from the GeneLogic's boxplot \ref{geneLogicBoxplot}.

\begin{figure}[!htb]
\centering
\includegraphics[width=130mm]{cleanedBoxplot.png}
\caption{cleanMD boxplot \label{cleanMDBoxplot}}
\end{figure}

\begin{figure}[!htb]
\centering
\includegraphics[width=130mm]{Gene_level_expressionBoxplot.png}
\caption{GeneLogic boxplot \label{geneLogicBoxplot}}
\end{figure}

\section{limma}

I've started reading the limma paper Simon sent, and my main conclusion is that I'm not ready. Instead I've been doing the background reading required to understand the paper -- I've read \& achieved a working understanding of False Discovry Rates (although I still don't understand the mathematical proof of the method) and started going through chapter 3 of Wasserman.

\section{Individual Specific Effects}

I've tried to reproduce prior work on Individual Specific Effects (ISE). I've been using linear regression as opposed to limma for now.

First, I computed 22k residuals, defined to be deviations from value predicted by best-fit line, produced for each transcript cluster by predicting transcript cluster intensities (dependent variable, outcome) from modal allele length (independent variable, feature) across all 35 individuals.

I plotted the results against Modal Allele length \ref{bloodAvgResidual}. As expected, the slope of the line is exactly 0. Worryingly, randomly permuting Modal Allele length \ref{permutedAvgResidual} gives average residuals which are visually just as "good/bad" as blood residuals.

I am yet to run t-tests on the set of residuals from which I compute the average residual (might actually have done it before our meeting at 4.00).

\begin{figure}[!htb]
\centering
\includegraphics[width=130mm]{bloodAvgResidual.png}
\caption{Average Blood Residual \label{bloodAvgResidual}}
\end{figure}


\begin{figure}[!htb]
\centering
\includegraphics[width=130mm]{permutedAvgResidual.png}
\caption{Permuted Blood Residual \label{permutedAvgResidual}}
\end{figure}

\printbibliography 
\end{document}
