{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'__pycache__'</li>\n",
       "\t<li>'analyseBlood.sh'</li>\n",
       "\t<li>'analyseLimma_files'</li>\n",
       "\t<li>'analyseLimma.aux'</li>\n",
       "\t<li>'analyseLimma.ipynb'</li>\n",
       "\t<li>'analyseLimma.log'</li>\n",
       "\t<li>'analyseLimma.out'</li>\n",
       "\t<li>'analyseLimma.pdf'</li>\n",
       "\t<li>'analyseLimma.tex'</li>\n",
       "\t<li>'CELfiles'</li>\n",
       "\t<li>'cleanDatasets.py'</li>\n",
       "\t<li>'cleaned'</li>\n",
       "\t<li>'cleanedLimma'</li>\n",
       "\t<li>'DMBDI genotyping blood DNA Glasgow data with DMBDI ID for John.txt'</li>\n",
       "\t<li>'experiments'</li>\n",
       "\t<li>'ISE'</li>\n",
       "\t<li>'JohnsGenes.csv'</li>\n",
       "\t<li>'krahe'</li>\n",
       "\t<li>'litReview.pdf'</li>\n",
       "\t<li>'MASTERAnonymizedClinicalDataset23Feb2012.txt'</li>\n",
       "\t<li>'masterBlood_files'</li>\n",
       "\t<li>'masterBlood.aux'</li>\n",
       "\t<li>'masterBlood.ipynb'</li>\n",
       "\t<li>'masterBlood.log'</li>\n",
       "\t<li>'masterBlood.out'</li>\n",
       "\t<li>'masterBlood.pdf'</li>\n",
       "\t<li>'masterBlood.tex'</li>\n",
       "\t<li>'microarraySanityCheck.ipynb'</li>\n",
       "\t<li>'normalise.aux'</li>\n",
       "\t<li>'normalise.ipynb'</li>\n",
       "\t<li>'normalise.log'</li>\n",
       "\t<li>'normalise.out'</li>\n",
       "\t<li>'normalise.pdf'</li>\n",
       "\t<li>'normalise.tex'</li>\n",
       "\t<li>'normalised'</li>\n",
       "\t<li>'normalisedFirstLine'</li>\n",
       "\t<li>'normalisedLastLines'</li>\n",
       "\t<li>'overleaf'</li>\n",
       "\t<li>'PGFFiles'</li>\n",
       "\t<li>'produceReport.sh'</li>\n",
       "\t<li>'progressReport.txt'</li>\n",
       "\t<li>'rawProbes.ipynb'</li>\n",
       "\t<li>'setup.R'</li>\n",
       "\t<li>'utils.py'</li>\n",
       "\t<li>'Wilf R file-clean.R'</li>\n",
       "\t<li>'WilfISEs.jpg'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '\\_\\_pycache\\_\\_'\n",
       "\\item 'analyseBlood.sh'\n",
       "\\item 'analyseLimma\\_files'\n",
       "\\item 'analyseLimma.aux'\n",
       "\\item 'analyseLimma.ipynb'\n",
       "\\item 'analyseLimma.log'\n",
       "\\item 'analyseLimma.out'\n",
       "\\item 'analyseLimma.pdf'\n",
       "\\item 'analyseLimma.tex'\n",
       "\\item 'CELfiles'\n",
       "\\item 'cleanDatasets.py'\n",
       "\\item 'cleaned'\n",
       "\\item 'cleanedLimma'\n",
       "\\item 'DMBDI genotyping blood DNA Glasgow data with DMBDI ID for John.txt'\n",
       "\\item 'experiments'\n",
       "\\item 'ISE'\n",
       "\\item 'JohnsGenes.csv'\n",
       "\\item 'krahe'\n",
       "\\item 'litReview.pdf'\n",
       "\\item 'MASTERAnonymizedClinicalDataset23Feb2012.txt'\n",
       "\\item 'masterBlood\\_files'\n",
       "\\item 'masterBlood.aux'\n",
       "\\item 'masterBlood.ipynb'\n",
       "\\item 'masterBlood.log'\n",
       "\\item 'masterBlood.out'\n",
       "\\item 'masterBlood.pdf'\n",
       "\\item 'masterBlood.tex'\n",
       "\\item 'microarraySanityCheck.ipynb'\n",
       "\\item 'normalise.aux'\n",
       "\\item 'normalise.ipynb'\n",
       "\\item 'normalise.log'\n",
       "\\item 'normalise.out'\n",
       "\\item 'normalise.pdf'\n",
       "\\item 'normalise.tex'\n",
       "\\item 'normalised'\n",
       "\\item 'normalisedFirstLine'\n",
       "\\item 'normalisedLastLines'\n",
       "\\item 'overleaf'\n",
       "\\item 'PGFFiles'\n",
       "\\item 'produceReport.sh'\n",
       "\\item 'progressReport.txt'\n",
       "\\item 'rawProbes.ipynb'\n",
       "\\item 'setup.R'\n",
       "\\item 'utils.py'\n",
       "\\item 'Wilf R file-clean.R'\n",
       "\\item 'WilfISEs.jpg'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '__pycache__'\n",
       "2. 'analyseBlood.sh'\n",
       "3. 'analyseLimma_files'\n",
       "4. 'analyseLimma.aux'\n",
       "5. 'analyseLimma.ipynb'\n",
       "6. 'analyseLimma.log'\n",
       "7. 'analyseLimma.out'\n",
       "8. 'analyseLimma.pdf'\n",
       "9. 'analyseLimma.tex'\n",
       "10. 'CELfiles'\n",
       "11. 'cleanDatasets.py'\n",
       "12. 'cleaned'\n",
       "13. 'cleanedLimma'\n",
       "14. 'DMBDI genotyping blood DNA Glasgow data with DMBDI ID for John.txt'\n",
       "15. 'experiments'\n",
       "16. 'ISE'\n",
       "17. 'JohnsGenes.csv'\n",
       "18. 'krahe'\n",
       "19. 'litReview.pdf'\n",
       "20. 'MASTERAnonymizedClinicalDataset23Feb2012.txt'\n",
       "21. 'masterBlood_files'\n",
       "22. 'masterBlood.aux'\n",
       "23. 'masterBlood.ipynb'\n",
       "24. 'masterBlood.log'\n",
       "25. 'masterBlood.out'\n",
       "26. 'masterBlood.pdf'\n",
       "27. 'masterBlood.tex'\n",
       "28. 'microarraySanityCheck.ipynb'\n",
       "29. 'normalise.aux'\n",
       "30. 'normalise.ipynb'\n",
       "31. 'normalise.log'\n",
       "32. 'normalise.out'\n",
       "33. 'normalise.pdf'\n",
       "34. 'normalise.tex'\n",
       "35. 'normalised'\n",
       "36. 'normalisedFirstLine'\n",
       "37. 'normalisedLastLines'\n",
       "38. 'overleaf'\n",
       "39. 'PGFFiles'\n",
       "40. 'produceReport.sh'\n",
       "41. 'progressReport.txt'\n",
       "42. 'rawProbes.ipynb'\n",
       "43. 'setup.R'\n",
       "44. 'utils.py'\n",
       "45. 'Wilf R file-clean.R'\n",
       "46. 'WilfISEs.jpg'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"__pycache__\"                                                       \n",
       " [2] \"analyseBlood.sh\"                                                   \n",
       " [3] \"analyseLimma_files\"                                                \n",
       " [4] \"analyseLimma.aux\"                                                  \n",
       " [5] \"analyseLimma.ipynb\"                                                \n",
       " [6] \"analyseLimma.log\"                                                  \n",
       " [7] \"analyseLimma.out\"                                                  \n",
       " [8] \"analyseLimma.pdf\"                                                  \n",
       " [9] \"analyseLimma.tex\"                                                  \n",
       "[10] \"CELfiles\"                                                          \n",
       "[11] \"cleanDatasets.py\"                                                  \n",
       "[12] \"cleaned\"                                                           \n",
       "[13] \"cleanedLimma\"                                                      \n",
       "[14] \"DMBDI genotyping blood DNA Glasgow data with DMBDI ID for John.txt\"\n",
       "[15] \"experiments\"                                                       \n",
       "[16] \"ISE\"                                                               \n",
       "[17] \"JohnsGenes.csv\"                                                    \n",
       "[18] \"krahe\"                                                             \n",
       "[19] \"litReview.pdf\"                                                     \n",
       "[20] \"MASTERAnonymizedClinicalDataset23Feb2012.txt\"                      \n",
       "[21] \"masterBlood_files\"                                                 \n",
       "[22] \"masterBlood.aux\"                                                   \n",
       "[23] \"masterBlood.ipynb\"                                                 \n",
       "[24] \"masterBlood.log\"                                                   \n",
       "[25] \"masterBlood.out\"                                                   \n",
       "[26] \"masterBlood.pdf\"                                                   \n",
       "[27] \"masterBlood.tex\"                                                   \n",
       "[28] \"microarraySanityCheck.ipynb\"                                       \n",
       "[29] \"normalise.aux\"                                                     \n",
       "[30] \"normalise.ipynb\"                                                   \n",
       "[31] \"normalise.log\"                                                     \n",
       "[32] \"normalise.out\"                                                     \n",
       "[33] \"normalise.pdf\"                                                     \n",
       "[34] \"normalise.tex\"                                                     \n",
       "[35] \"normalised\"                                                        \n",
       "[36] \"normalisedFirstLine\"                                               \n",
       "[37] \"normalisedLastLines\"                                               \n",
       "[38] \"overleaf\"                                                          \n",
       "[39] \"PGFFiles\"                                                          \n",
       "[40] \"produceReport.sh\"                                                  \n",
       "[41] \"progressReport.txt\"                                                \n",
       "[42] \"rawProbes.ipynb\"                                                   \n",
       "[43] \"setup.R\"                                                           \n",
       "[44] \"utils.py\"                                                          \n",
       "[45] \"Wilf R file-clean.R\"                                               \n",
       "[46] \"WilfISEs.jpg\"                                                      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "character(0)\n"
     ]
    }
   ],
   "source": [
    "# load the oligo library\n",
    "library(oligo)\n",
    "\n",
    "# Read in the CEL files in the directory\n",
    "exampleCEL = file.path(getwd(), \"affy_v4_example.CEL\")\n",
    "\n",
    "celFiles <- list.celfiles(exampleCEL)\n",
    "\n",
    "list.files()\n",
    "\n",
    "print(celFiles)\n",
    "\n",
    "#affyRaw <- read.celfiles(celFiles)\n",
    "\n",
    "# Load up-to-date annotations for human exon gene microarray.\n",
    "#library(pd.huex.1.0.st.v2)\n",
    "\n",
    "# RMA-normalise into `transcription clusters` using the loaded annotations.\n",
    "#core <- rma(affyRaw, target=\"core\")\n",
    "#probeset <- rma(affyRaw, target=\"probeset\")\n",
    "#full <- rma(affyRaw, target=\"full\")\n",
    "#extended <- rma(affyRaw, target=\"extended\")\n",
    "\n",
    "# Finally, save the data to an output file to be used by other programs, etc (Data will be log2 transformed and normalized)\n",
    "#write.exprs(core,file=\"core\")\n",
    "\n",
    "# In case we decide it's desirable to do so\n",
    "#write.exprs(probeset,file=\"probeset\")\n",
    "#write.exprs(full, file=\"full\")\n",
    "#write.exprs(extended, file=\"extended\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.3.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
